test/primitives/blocks_looks.mjs

```diff
@@ -0,0 +1,596 @@
+            const target = new RenderedTarget(sprite, rt);
+            rt.addTarget(target);
+
+            const retVal = await rt.execBlockPrimitive(target.id, 'looks_say', {MESSAGE: 'Hello World'}, 'test_token');
+
+            // the Scratch.looks custom state get's a target's bubble-related state
+            expect(target.getCustomState('Scratch.looks').type).to.equal('say');
+            expect(target.getCustomState('Scratch.looks').text).to.equal('Hello World');
+            expect(retVal).to.equal(undefined);
+        });
+
+        it('Think', async () => {
+            const rt = new Runtime();
+            const sprite = new Sprite(null, rt);
+            const target = new RenderedTarget(sprite, rt);
+            rt.addTarget(target);
+
+            const retVal = await rt.execBlockPrimitive(target.id, 'looks_think', {MESSAGE: 'Hello World'}, 'test_token');
+
+            expect(target.getCustomState('Scratch.looks').type).to.equal('think');
+            expect(target.getCustomState('Scratch.looks').text).to.equal('Hello World');
+            expect(retVal).to.equal(undefined);
+        });
+
+        // TODO: Say and Think for Seconds
+
+        it('Show', async () => {
+            const rt = new Runtime();
+            const sprite = new Sprite(null, rt);
+            const target = new RenderedTarget(sprite, rt);
+            rt.addTarget(target);
+
+            const retVal = await rt.execBlockPrimitive(target.id, 'looks_show', {}, 'test_token');
+
+            expect(target.visible).to.equal(true);
+            expect(retVal).to.equal(undefined);
+        });
+
+        it('Hide', async () => {
+            const rt = new Runtime();
+            const sprite = new Sprite(null, rt);
+            const target = new RenderedTarget(sprite, rt);
+            rt.addTarget(target);
+
+            const retVal = await rt.execBlockPrimitive(target.id, 'looks_hide', {}, 'test_token');
+
+            expect(target.visible).to.equal(false);
+            expect(retVal).to.equal(undefined);
+        });
+
+        it('Set Costume To From Index', async () => {
+            const rt = new Runtime();
+            const sprite = new Sprite(null, rt);
+            const target = new RenderedTarget(sprite, rt);
+            rt.addTarget(target);
+
+            const costumeCatWalk = {
+                asset: null,
+                assetId: null,
+                skinId: null,
+                name: 'cat-walk',
+                bitmapResolution: null,
+                rotationCenterX: 0,
+                rotationCenterY: 0
+            };
+
+            const costumeCatRun = {
+                asset: null,
+                assetId: null,
+                skinId: null,
+                name: 'cat-run',
+                bitmapResolution: null,
+                rotationCenterX: 0,
+                rotationCenterY: 0
+            };
+
+            const costumeCatFly = {
+                asset: null,
+                assetId: null,
```

Comment:
Can this code be refactored into the before() hook like you did with the integration tests?